integration/disk_cleanup_deletes_inactive_directories_test.go

```diff
@@ -75,7 +76,29 @@ func TestDiskCleansupInactiveFilesets(t *testing.T) {
 	require.NoError(t, err)
 	testSetup.db.AssignShardSet(shardSet)
 
+	// So some gnarly stuff here to make the testing package a bit more synchronous
+	var fsCleanupErr chan error
+	var nsCleanupErr chan error
 	// Check if files have been deleted
 	waitTimeout := 30 * time.Second
-	require.NoError(t, waitUntilFilesetsCleanedUp(filePathPrefix, md.ID(), extraShard.ID(), waitTimeout))
+
+	go func() {
+		fsCleanupErr <- waitUntilFilesetsCleanedUp(filePathPrefix,
+			testSetup.db.Namespaces(), extraShard.ID(), waitTimeout)
+		require.NoError(t, <-fsCleanupErr)
+	}()
+
+	// Assigning a shardset of length one will mean there are extra namespaces
+	// that will need to be removed via cleanup
+	shardSet, err = sharding.NewShardSet(shards[:1], shardSet.HashFn())
+	require.NoError(t, err)
+	testSetup.db.AssignShardSet(shardSet)
+
+	// Needs an extra 30 seconds due to race conditions
+	waitTimeout = 160 * time.Second
+	go func() {
+		nsCleanupErr <- waitUntilNamespacesCleanedUp(testSetup, filePathPrefix,
+			testNamespaces[1], waitTimeout)
+		require.NoError(t, <-nsCleanupErr)
+	}()
```

Comment:
You'll need to wait for both the goroutines that got launched here to finish before the test finishes, otherwise it might look like it passed when some of those routines haven't finished yet.

Also is there a reason you're using channels here when you read straight off the channel afterwards?